### PR TITLE
Add Project boilerplate

### LINKED ISSUE #110: Add the project module boilerplate
## Description
We need to add the boilerplate needed for the module responsible for adding new projects. We just need the boilerplate, no need to have concrete actions or handling.

One thing that needs to be added to the state is a `editableProject` type, similar to the editable time entry. That property is the one that will control the visibility of the project view

## Definition of done
- [x] The ProjectReducer is added
- [x] The ProjectState is added
- [x] The ProjectViewController is added
- [x] The coordinator boilerplate needed is added

### PR DESCRIPTION
**Pull request by heytherewill**
### Summary
This adds the basic boilerplate needed for work on the project module to start

### Technical considerations
I've added a little enum to avoid the string comparison that was happening in that view. I don't fully get why there's a list and all those casts going on when there could simply statically typed properties; I'll wait for @Odrakir to chime in and if there's no reason I'll just change that as well.

### Relationships
Closes #110 

### Pull Request checklist
- [X] The code does not hinder the app's performance
- [X] The code does not introduce regressions
- [X] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [X] The code used custom templates

### Pointers
Simple stuff and boilerplate, nothing too exciting to see here

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
Anyone, anytime

### CODE REVIEW

**Comment by Odrakir** [comment]
The enum is fine. I don't really like the name of the feature: "Project"... Project what? Selection? Creation?

**Comment by heytherewill** [comment]
@Odrakir I does both. Since `StartEdit` is taken by the most prominent feature of the app (time entries), naming this `StartEditProject` is just needlessly verbose. There's no feature for selecting project in the new app, since the start/edit view will be unified.

**Comment by Odrakir** [comment]
`ProjectManagment` then?

File: Modules/Timer/Timer/Common/TimerCoordinator.swift [hunk 1]
```diff
@@ -8,15 +8,19 @@ public final class TimerCoordinator: BaseCoordinator {
...
-    public init(store: Store<TimerState, TimerAction>,
-                timeLogCoordinator: TimeEntriesLogCoordinator,
-                startEditCoordinator: StartEditCoordinator,
-                runningTimeEntryCoordinator: RunningTimeEntryCoordinator) {
+    public init(
+        store: Store<TimerState, TimerAction>,
+        timeLogCoordinator: TimeEntriesLogCoordinator,
+        startEditCoordinator: StartEditCoordinator,
+        runningTimeEntryCoordinator: RunningTimeEntryCoordinator,
+        projectCoordinator: ProjectCoordinator) {
```
Comment (sofokli1) [inline Modules/Timer/Timer/Common/TimerCoordinator.swift:18]: a new line here would make this more readable

File: Modules/Timer/Timer/TimerFeature.swift [hunk 2]
```diff
@@ -10,38 +10,54 @@ public func createTimerReducer(repository: Repository, time: Time) -> Reducer<Ti
...
-            timeLogCoordinator: (features["log"]!.mainCoordinator(store: store) as? TimeEntriesLogCoordinator)!,
-            startEditCoordinator: (features["startEdit"]!.mainCoordinator(store: store) as? StartEditCoordinator)!,
-            runningTimeEntryCoordinator: (features["runningTimeEntry"]!.mainCoordinator(store: store) as? RunningTimeEntryCoordinator)!
+            timeLogCoordinator: features[.timeEntriesLog]!.mainCoordinator(store: store) as! TimeEntriesLogCoordinator,
+            startEditCoordinator: features[.startEdit]!.mainCoordinator(store: store) as! StartEditCoordinator,
+            runningTimeEntryCoordinator: features[.runningTimeEntry]!.mainCoordinator(store: store) as! RunningTimeEntryCoordinator,
+            projectCoordinator: features[.project]!.mainCoordinator(store: store) as! ProjectCoordinator
         )
     }
+    // swiftlint:enable force_cast
```
Comment (sofokli1) [inline Modules/Timer/Timer/TimerFeature.swift:60]: shouldn't we keep all the linter comments at the top of the file, as we do in most files, just so we know where to look for these?

**Comment by heytherewill** [comment]
@Odrakir This is a great name for the feature that lists projects and allows you to delete them in batch/see them all, like in webapp. I think Project is good enough as is for an add feature

**Comment by Odrakir** [comment]
`AddProject` then? I'm saying this because `Project` is the model, the feature is supposed to do something with it, I think it could be confusing.

**Pull request merged**